installer/bootenv.go

```diff
@@ -91,7 +98,44 @@ func (e *UBootEnv) ReadEnv(names ...string) (BootVars, error) {
 	return vars, err
 }
 
+// ProbeSeparator tries writing 'mender_uboot_separator' to the U-Boot
+// environment using whitespace as a separator. If this write is succesful it
+// means that we are on a legacy U-Boot implementation, which supports the
+// 'key<whitespace>value' syntax. If not, then use the new '=' separator.
+func (e *UBootEnv) ProbeSeparator() (string, error) {
+	// Try writing using the old separator syntax
+	err := e.writeEnvImpl(BootVars{
+		menderUBootSeparatorProbe: "1",
+	}, " ")
+	v, err := e.ReadEnv(menderUBootSeparatorProbe)
+	if err != nil {
+		return uBootEnvStandardSeparator, err
+	}
+	if v[menderUBootSeparatorProbe] == "1" {
+		// Clean variable from the environment
+		err = e.writeEnvImpl(BootVars{
+			menderUBootSeparatorProbe: "",
+		}, " ")
+		return uBootEnvLegacySeparator, err
+	}
+	return uBootEnvStandardSeparator, nil
+}
+
+// WriteEnv attempts to write the given 'BootVars' to the bootloader
+// environment.
 func (e *UBootEnv) WriteEnv(vars BootVars) error {
+	// Probe for the separator used by U-Boot. Newer versions support '=',
+	// and libubootenv only supports '=' as the separator, older versions
+	// only support ' '
+	separator, err := e.ProbeSeparator()
+	if err != nil {
+		log.Errorf("Failed to probe the U-Boot environment for which separator to use. Got error: %v, Going to try writing the U-Boot environment using the separator: %q", err, separator)
```

Comment:
In general I think we should use `%s` and `err.Error()`. We've had so many cases of `%v` printing the wrong thing because there is no type checking.